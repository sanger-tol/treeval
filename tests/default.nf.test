nextflow_pipeline {

    name "Test TreeVal Full pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "full"


    test("Full run") {

        setup {
            println "Downloading BUSCO test data..."
            def busco_test_url = "https://tolit.cog.sanger.ac.uk/test-data/resources/treeval/TreeValTinyData/busco.tar.gz"
            def busco_test_command = ['bash', '-c', "curl -L --retry 5 --retry-delay 10 ${busco_test_url} | tar xzf - -C ${launchDir}"]
            def busco_test_proc = busco_test_command.execute()
            busco_test_proc.waitFor()

            if (busco_test_proc.exitValue() != 0) {
                throw new RuntimeException("Error - failed to download busco data: ${busco_test_proc.err.text}")
            }
        }

        when {
            params {
                input = "${projectDir}/assets/TreeValTinyFullTest.yaml"
                outdir = "$outputDir"
                mode = "FULL"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignore: ['tests/.nftignore'])

            // def mcool_path = getAllFilesFromDir("${params.outdir}/hic_files/", include: ["*.mcool"])

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),

                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/treeval_software_versions.yml"),

                    // All stable path name, with a relative path
                    stable_name,

                    // All files with stable contents
                    stable_path

                ).match() }
            )
        }
        cleanup {
            new File("${launchDir}/busco")
        }
    }
}
