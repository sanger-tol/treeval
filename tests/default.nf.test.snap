{
    "Full run": {
        "content": [
            101,
            {
                "BAMTOBED_SORT": {
                    "samtools": 1.17,
                    "bedtools": "2.31.0"
                },
                "BED2BW_NORMAL": {
                    "ucsc": 469
                },
                "BEDTOOLS_BAMTOBED": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_GENOMECOV": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_MAKEWINDOWS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_MAP": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_MERGE_MIN": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "BUSCO_BUSCO": {
                    "busco": "5.7.1"
                },
                "BWAMEM2_INDEX": {
                    "bwamem2": "2.2.1"
                },
                "CAT_CAT": {
                    "pigz": "2.3.4"
                },
                "CONCAT_BLOCKS": {
                    "coreutils": 9.1
                },
                "COOLER_CLOAD": {
                    "cooler": "0.9.2"
                },
                "COOLER_ZOOMIFY": {
                    "cooler": "0.9.2"
                },
                "CRAM_FILTER_ALIGN_BWAMEM2_FIXMATE_SORT": {
                    "samtools": 1.17,
                    "bwa-mem2": null
                },
                "EXTRACT_REPEAT": {
                    "perl": "5, version 26, subversion 2 (v5.26.2) built for x86_64-linux-thread-multi Copyright 1987-2018, Larry Wall Perl may be copied only under the terms of either the Artistic License or the GNU General Public License, which may be found in the Perl 5 source kit. Complete documentation for Perl, including FAQ lists, should be found on this system using \"man perl\" or \"perldoc perl\". If you have access to the Internet, point your browser at http://www.perl.org/, the Perl Home Page.",
                    "extract_repeat.pl": 1.0
                },
                "EXTRACT_TELO": {
                    "coreutils": 9.1
                },
                "FASTK_FASTK": {
                    "fastk": "f18a4e6d2207539f7b84461daebc54530a9559b0"
                },
                "FIND_HALF_COVERAGE": {
                    "python": "Python 3.9.1",
                    "findHalfcoverage.py": 1.0
                },
                "FIND_TELOMERE_REGIONS": {
                    "find_telomere": null
                },
                "FIND_TELOMERE_WINDOWS": {
                    "telomere": 1.0
                },
                "GAWK_CLEAN_TELOMERE": {
                    "gawk": "5.3.0"
                },
                "GAWK_EXTRACT_BUSCOGENE": {
                    "gawk": "5.3.0"
                },
                "GAWK_GAP_LENGTH": {
                    "gawk": "5.3.0"
                },
                "GAWK_MAP_TELO": {
                    "gawk": "5.3.0"
                },
                "GAWK_REFORMAT_INTERSECT": {
                    "gawk": "5.3.0"
                },
                "GAWK_RENAME_IDS": {
                    "gawk": "5.3.0"
                },
                "GAWK_REPLACE_DOTS": {
                    "gawk": "5.3.0"
                },
                "GAWK_SPLIT_DIRECTIONS": {
                    "gawk": "5.3.0"
                },
                "GAWK_UPPER_SEQUENCE": {
                    "gawk": "5.3.0"
                },
                "GENERATE_CRAM_CSV": {
                    "generate_cram_csv": 1.1,
                    "samtools": 1.17
                },
                "GET_LARGEST_SCAFFOLD": {
                    "coreutils": 9.1
                },
                "GET_MIN_MAX_PUNCHES": {
                    "coreutils": 9.1
                },
                "GET_PAIRED_CONTACT_BED": {
                    "bed_to_contacts": "1.0.0"
                },
                "GNU_SORT_A": {
                    "coreutils": 9.3
                },
                "GNU_SORT_B": {
                    "coreutils": 9.3
                },
                "GNU_SORT_BED": {
                    "coreutils": 9.3
                },
                "GNU_SORT_C": {
                    "coreutils": 9.3
                },
                "GNU_SORT_COVBED": {
                    "coreutils": 9.3
                },
                "GRAPH_OVERALL_COVERAGE": {
                    "perl": "5, version 26, subversion 2 (v5.26.2) built for x86_64-linux-thread-multi Copyright 1987-2018, Larry Wall Perl may be copied only under the terms of either the Artistic License or the GNU General Public License, which may be found in the Perl 5 source kit. Complete documentation for Perl, including FAQ lists, should be found on this system using \"man perl\" or \"perldoc perl\". If you have access to the Internet, point your browser at http://www.perl.org/, the Perl Home Page.",
                    "graph_overall_coverage.pl": 1.0
                },
                "MAKECMAP_CMAP2BED": {
                    "python": "Python 3.9.1",
                    "cmap2bed.py": 1.0
                },
                "MAKECMAP_FA2CMAPMULTICOLOR": {
                    "perl": "5, version 26, subversion 2 (v5.26.2) built for x86_64-linux-thread-multi Copyright 1987-2018, Larry Wall Perl may be copied only under the terms of either the Artistic License or the GNU General Public License, which may be found in the Perl 5 source kit. Complete documentation for Perl, including FAQ lists, should be found on this system using \"man perl\" or \"perldoc perl\". If you have access to the Internet, point your browser at http://www.perl.org/, the Perl Home Page.",
                    "fa2cmap_multi_color.pl": "fa2cmap_multi_color.pl 7821 2018-08-09 17:30:15Z twang"
                },
                "MAKECMAP_RENAMECMAPIDS": {
                    "perl": "5, version 26, subversion 2 (v5.26.2) built for x86_64-linux-thread-multi Copyright 1987-2018, Larry Wall Perl may be copied only under the terms of either the Artistic License or the GNU General Public License, which may be found in the Perl 5 source kit. Complete documentation for Perl, including FAQ lists, should be found on this system using \"man perl\" or \"perldoc perl\". If you have access to the Internet, point your browser at http://www.perl.org/, the Perl Home Page.",
                    "rename_cmapids.pl": 1.0
                },
                "MERQURYFK_MERQURYFK": {
                    "fastk": "f18a4e6d2207539f7b84461daebc54530a9559b0",
                    "merquryfk": "8ae344092df5dcaf83cfb7f90f662597a9b1fc61",
                    "r": "4.2.0"
                },
                "MINIMAP2_ALIGN": {
                    "minimap2": "2.28-r1209",
                    "samtools": 1.2
                },
                "MINIPROT_INDEX": {
                    "miniprot": "0.11-r234"
                },
                "MUMMER": {
                    "mummer": 3.23
                },
                "PAF2BED": {
                    "paf_to_bed": "1.0.0",
                    "coreutils": 9.1
                },
                "PAFTOOLS_SAM2PAF": {
                    "paftools.js": null
                },
                "PRETEXTMAP_HIGHRES": {
                    "PretextGraph": "PretextMap Version 0.0.9",
                    "PretextMap": "0.1.9",
                    "samtools": 1.21
                },
                "PRETEXTMAP_STANDRD": {
                    "PretextGraph": "PretextMap Version 0.0.9",
                    "PretextMap": "0.1.9",
                    "samtools": 1.21
                },
                "PRETEXT_INGEST_HIRES": {
                    "PretextGraph": "0.0.9",
                    "PretextMap": "0.1.9",
                    "bigWigToBedGraph": 447
                },
                "PRETEXT_INGEST_SNDRD": {
                    "PretextGraph": "0.0.9",
                    "PretextMap": "0.1.9",
                    "bigWigToBedGraph": 447
                },
                "SAMTOOLS_MARKDUP": {
                    "samtools": 1.21
                },
                "SAMTOOLS_MERGE": {
                    "samtools": 1.21
                },
                "SELFCOMP_ALIGNMENTBLOCKS": {
                    "python": "Python 3.10.8",
                    "pandas": null,
                    "pybedtools": null,
                    "build_alignment_block.py": 1.0
                },
                "SELFCOMP_MAPIDS": {
                    "python": "Python 3.9.1",
                    "mapids.py": 1.0
                },
                "SELFCOMP_MUMMER2BED": {
                    "python": "Python 3.9.1",
                    "mummer2bed.py": 1.0
                },
                "SELFCOMP_SPLITFASTA": {
                    "perl": "5, version 32, subversion 1 (v5.32.1) built for x86_64-linux-thread-multi Copyright 1987-2021, Larry Wall Perl may be copied only under the terms of either the Artistic License or the GNU General Public License, which may be found in the Perl 5 source kit. Complete documentation for Perl, including FAQ lists, should be found on this system using \"man perl\" or \"perldoc perl\". If you have access to the Internet, point your browser at http://www.perl.org/, the Perl Home Page.",
                    "perl-bioperl": "1.7.8-1",
                    "split_genomes_for_ensembl.pl": 1.0
                },
                "SEQKIT_SPLIT_QUERY": {
                    "seqkit": "2.9.0"
                },
                "SEQKIT_SPLIT_REF": {
                    "seqkit": "2.9.0"
                },
                "SEQTK_CUTN": {
                    "seqtk": "1.4-r122"
                },
                "SNAPSHOT_SRES": {
                    "pretextsnapshot": "0.0.4"
                },
                "TABIX_BGZIPTABIX": {
                    "tabix": 1.2
                },
                "UCSC_BEDGRAPHTOBIGWIG": {
                    "ucsc": 469
                },
                "UCSC_BEDTOBIGBED": {
                    "ucsc": 447
                },
                "WINDOWMASKER_MKCOUNTS": {
                    "windowmasker": "1.0.0"
                },
                "WINDOWMASKER_USTAT": {
                    "windowmasker": "1.0.0"
                },
                "Workflow": {
                    "sanger-tol/treeval": "v1.4.0"
                }
            },
            [
                [
                    ".OscheiusSUBSET_1_fk.ktab.1:md5,ef0a313b59e7a4889a637361ce0f6430",
                    ".OscheiusSUBSET_1_fk.ktab.2:md5,efe72285706d0943f56f4ea46910c3c3",
                    ".OscheiusSUBSET_1_fk.ktab.3:md5,665c0cf6eba82ae82906fa79e050c874",
                    ".OscheiusSUBSET_1_fk.ktab.4:md5,ddfc0806ab2eebbdebd3069a67ac4b83",
                    ".OscheiusSUBSET_1_fk.ktab.5:md5,270f7f573ad97bf388fa89eb3860e27b",
                    ".OscheiusSUBSET_1_fk.ktab.6:md5,fa104e8d64dcba20b661b01506e422aa",
                    ".OscheiusSUBSET_1_fk.ktab.7:md5,b04b3ae5aed2828d1777f4c3b838a7d6",
                    ".OscheiusSUBSET_1_fk.ktab.8:md5,f4c7206b3c603c56c8ab6d1fa90aabbf",
                    "OscheiusSUBSET_1.mcool:md5,c931f41c68967eb7f907d3df3a4d3d59",
                    "OscheiusSUBSET_1_fk.ktab:md5,ccf29ca221f52ac1595846ee7a8a25ee",
                    "OscheiusSUBSET_1_hr.pretext:md5,19de425236696fe45f5258114bc54209",
                    "OscheiusSUBSET_1_normal.pretext:md5,d2af1377ff53319168e9a0795d5d682a",
                    "OscheiusSUBSET_1_normalFullMap.png:md5,8d03bdd32da5a8bcf1b7dd5f376f8db1",
                    "OscheiusSUBSET_1_repeat_density.bigWig:md5,779ad07ceefaca4657090c9f0322ddfd",
                    "OscheiusSUBSET_1_repeat_mk.bed:md5,5205d07d425f795e435916ab73553a9a",
                    "OscheiusSUBSET_1_telomere.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "OscheiusSUBSET_1_telomere.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "bAytFul3_GRIT-1165_gap.bed:md5,e00e46e71c89450392959fbc5321ea66"
                ],
                "hic_files/grTriPseu1_1.gap.bedgraph",
                "hic_files/grTriPseu1_1.grTriPseu1.spectra-cn.ln.png",
                "hic_files/grTriPseu1_1.mcool",
                "hic_files/grTriPseu1_13P_telomere.bed",
                "hic_files/grTriPseu1_13P_telomere.bedgraph",
                "hic_files/grTriPseu1_15P_telomere.bed",
                "hic_files/grTriPseu1_15P_telomere.bedgraph",
                "hic_files/grTriPseu1_1_coverage_normal.bigWig",
                "hic_files/grTriPseu1_1_gap.bed",
                "hic_files/grTriPseu1_1_hr.pretext",
                "hic_files/grTriPseu1_1_normal.pretext",
                "hic_files/grTriPseu1_1_normalFullMap.png",
                "hic_files/grTriPseu1_1_repeat_density.bigWig",
                "hic_files/grTriPseu1_1_repeat_mk.bed",
                "hic_files/grTriPseu1_1_telomere.bed",
                "hic_files/grTriPseu1_1_telomere.bedgraph",
                "pipeline_info",
                "pipeline_info/treeval_software_versions.yml",
                [
                    "BSPQI.bigBed:md5,8d5f8fce6db304018dabe8f2721bb7e5",
                    "BSSSI.bigBed:md5,83d7b0324d8f6ae60a17356d0545d15a",
                    "DLE1.bigBed:md5,81cd559d4b8e0e1406142823006fe111",
                    "OscheiusSUBSET_1_len_sorted.genome:md5,6a1f75afdc99390c150a9abe204e856b",
                    "OscheiusTipulae.ASM1342590v1_cdna.bigBed:md5,f372288fa14cb24372dd0528e3e1336d",
                    "bAytFul3_GRIT-1165.fa.fai:md5,642254b59465adeae1b7fc0dcd74980c",
                    "bAytFul3_GRIT-1165.unsorted.genome:md5,7fe346bf17ad7c98c62dc736c9d05329",
                    "bCucCan1_synteny_Oscheius_SUBSET.paf:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "bGalGal1_synteny_Oscheius_SUBSET.paf:md5,608f79576de572440b94f670abc21b6d",
                    [
                        "OscheiusTipulae.ASM1342590v1_cdna_punchlist.bed:md5,5001ad58ba3d9a415066db4922f29444"
                    ],
                    "repden_OscheiusSUBSET_1.bed.gz:md5,5205d07d425f795e435916ab73553a9a",
                    "repden_OscheiusSUBSET_1.bed.gz.csi:md5,02606032aa5ee3ec40569a6136a66616",
                    "telo_OscheiusSUBSET_1.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "telo_OscheiusSUBSET_1.bed.gz.csi:md5,a76ffd9c2c19bc3100eecc93162f31d9"
                ],
                "BSPQI.bigBed:md5,8d5f8fce6db304018dabe8f2721bb7e5",
                "BSSSI.bigBed:md5,83d7b0324d8f6ae60a17356d0545d15a",
                "DLE1.bigBed:md5,81cd559d4b8e0e1406142823006fe111",
                "treeval_upload/LaetiporusSulphureus.gfLaeSulp1_cdna.bigBed",
                "treeval_upload/LaetiporusSulphureus_synteny_grTriPseu1.paf",
                "treeval_upload/gap_grTriPseu1_1.bed.gz",
                "treeval_upload/gap_grTriPseu1_1.bed.gz.csi",
                "treeval_upload/grTriPseu1_1.bed.gz",
                "treeval_upload/grTriPseu1_1.bed.gz.tbi",
                "treeval_upload/grTriPseu1_13P.bed.gz",
                "treeval_upload/grTriPseu1_13P.bed.gz.tbi",
                "treeval_upload/grTriPseu1_15P.bed.gz",
                "treeval_upload/grTriPseu1_15P.bed.gz.tbi",
                "treeval_upload/grTriPseu1_1_buscogene.bigBed",
                "treeval_upload/grTriPseu1_1_len_sorted.genome",
                "treeval_upload/grTriPseu1_1_selfcomp.bigBed",
                [
                    "OscheiusTipulae.ASM1342590v1_cdna_punchlist.bed:md5,5001ad58ba3d9a415066db4922f29444"
                ],
                "treeval_upload/punchlists/LaetiporusSulphureus.gfLaeSulp1_cdna_punchlist.bed",
                "treeval_upload/punchlists/halfcoverage.bed",
                "treeval_upload/punchlists/zerodepth.bed",
                "treeval_upload/repden_grTriPseu1_1.bed.gz",
                "treeval_upload/repden_grTriPseu1_1.bed.gz.csi"
            ],
            [
                "grTriPseu1_1.gap.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_1.grTriPseu1.spectra-cn.ln.png:md5,cddd75c3905022d973fc35151d2ebfcd",
                "grTriPseu1_13P_telomere.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_13P_telomere.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_15P_telomere.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_15P_telomere.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_1_coverage_normal.bigWig:md5,175b646ebabb2a56f58a6b6abdfa168f",
                "grTriPseu1_1_gap.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_1_repeat_density.bigWig:md5,3f66a9152d793a62f877b733c2336dfd",
                "grTriPseu1_1_repeat_mk.bed:md5,ad9eacea65665b8ae884df2f3075dba6",
                "grTriPseu1_1_telomere.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_1_telomere.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e",
                "BSPQI.bigBed:md5,d33e1c0b62c37df2612559fe4d471c56",
                "BSSSI.bigBed:md5,b43f1a087bc134a16444705e3e727b7b",
                "DLE1.bigBed:md5,173ab95349213d3a76d0fc6e1f1d559a",
                "LaetiporusSulphureus.gfLaeSulp1_cdna.bigBed:md5,853c2e0c60be0345947cc93a807f1f9e",
                "LaetiporusSulphureus_synteny_grTriPseu1.paf:md5,08742c3817f0da391c7a6e78b5c0f83b",
                "gap_grTriPseu1_1.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "gap_grTriPseu1_1.bed.gz.csi:md5,a76ffd9c2c19bc3100eecc93162f31d9",
                "grTriPseu1_1.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_1.bed.gz.tbi:md5,ec06685d85d15d8f118a789a9ccc42ad",
                "grTriPseu1_13P.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_13P.bed.gz.tbi:md5,ec06685d85d15d8f118a789a9ccc42ad",
                "grTriPseu1_15P.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "grTriPseu1_15P.bed.gz.tbi:md5,ec06685d85d15d8f118a789a9ccc42ad",
                "grTriPseu1_1_buscogene.bigBed:md5,894f75db52965f97c43f5a98f1c79aff",
                "grTriPseu1_1_len_sorted.genome:md5,150bfa0a8719e091014111b313b42395",
                "grTriPseu1_1_selfcomp.bigBed:md5,39dc23e73bf893b2f258703fb44f2c5d",
                "LaetiporusSulphureus.gfLaeSulp1_cdna_punchlist.bed:md5,e76395a1251ae1017220ec751f856d1d",
                "halfcoverage.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "zerodepth.bed:md5,eb0d0b05b3961a93e67d03df14d4d751",
                "repden_grTriPseu1_1.bed.gz:md5,ad9eacea65665b8ae884df2f3075dba6",
                "repden_grTriPseu1_1.bed.gz.csi:md5,736b2f22beded41899bef7e07b96fb88"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-07-31T18:28:46.449138"
    }
}